checks/evaluation/pinned_dependencies.go

```diff
@@ -110,25 +139,43 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 	}
 
 	// Docker downloads.
-	dockerDownloadScore, err := createReturnForIsDockerfileFreeOfInsecureDownloads(pr, dl)
+	dockerDownloadScore, err := createReturnForIsDockerfileFreeOfInsecureDownloads(d, dl)
 	if err != nil {
 		return checker.CreateRuntimeErrorResult(name, err)
 	}
 
 	// Script downloads.
-	scriptScore, err := createReturnForIsShellScriptFreeOfInsecureDownloads(pr, dl)
+	scriptScore, err := createReturnForIsShellScriptFreeOfInsecureDownloads(d, dl)
 	if err != nil {
 		return checker.CreateRuntimeErrorResult(name, err)
 	}
 
-	// Scores may be inconclusive.
-	actionScore = maxScore(0, actionScore)
-	dockerFromScore = maxScore(0, dockerFromScore)
-	dockerDownloadScore = maxScore(0, dockerDownloadScore)
-	scriptScore = maxScore(0, scriptScore)
+	// Pip installs.
+	pipScore, err := createReturnForIsPipInstallPinned(pr, dl)
+	if err != nil {
+		return checker.CreateRuntimeErrorResult(name, err)
+	}
+
+	// Npm installs.
+	npmScore, err := createReturnForIsNpmInstallPinned(pr, dl)
+	if err != nil {
+		return checker.CreateRuntimeErrorResult(name, err)
+	}
+
+	// Choco installs.
+	chocoScore, err := createReturnForIsChocoInstallPinned(pr, dl)
+	if err != nil {
+		return checker.CreateRuntimeErrorResult(name, err)
+	}
+
+	// Go installs.
+	goScore, err := createReturnForIsGoInstallPinned(pr, dl)
+	if err != nil {
+		return checker.CreateRuntimeErrorResult(name, err)
+	}
 
 	score := checker.AggregateScores(actionScore, dockerFromScore,
```

Comment:
Does including `pipScore`, `npmScore`, etc. double-count a dependency that's already been counted in `scriptScore`?

Reply:
No, because `scriptScore` counts only `downloadThenRun` commands, which is looks for `curl <pkg> && bash <pkg>` snippets in scripts.

Reply:
Would you like me to create a test to demonstrate that?